### PR TITLE
Add examples with Intel optimizations 

### PR DESCRIPTION
**Pull request by hshen14**
Per comments from https://github.com/huggingface/diffusers/pull/1499, this PR is to add examples with Intel optimizations for fine-tuning and inference. Bfloat16 fine-tuning is enabled for textual_inversion, while Bfloat16 inference is generally applicable for Stable Diffusion.

### CODE REVIEW

**Comment by hshen14** [comment]
@patrickvonplaten @anton-l @patil-suraj  please review. Thanks.

File: examples/README.md [hunk 1]
```diff
@@ -52,6 +52,10 @@ For such examples, we are more lenient regarding the philosophy defined above an
 Examples that are useful for the community, but are either not yet deemed popular or not yet following our above philosophy should go into the [community examples](https://github.com/huggingface/diffusers/tree/main/examples/community) folder. The community folder therefore includes training examples and inference pipelines.
 **Note**: Community examples can be a [great first contribution](https://github.com/huggingface/diffusers/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22) to show to the community how you like to use `diffusers` 🪄.
 
+## Research Projects
+
+We also provide **research_projects** examples that can be used to accelerate the training (fine-tuning) and inference. These examples are useful and offer the extended capablities which are complementary to the official examples. You may refer to [research_projects](https://github.com/huggingface/diffusers/tree/main/examples/research_projects) for details.
```
Comment (patrickvonplaten) [inline examples/README.md:57]:
```suggestion
We also provide **research_projects** examples that are maintained by the community as defined in the respective research project folders. These examples are useful and offer the extended capabilities which are complementary to the official examples. You may refer to [research_projects](https://github.com/huggingface/diffusers/tree/main/examples/research_projects) for details.
```

File: examples/research_projects/intel_opts/inference_bf16.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import torch
```
Comment (patrickvonplaten) [inline examples/research_projects/intel_opts/inference_bf16.py:1]: Could we add a README.md to `research_projects/intel_opts`?

File: examples/README.md (hunk 1 above)
Comment (hshen14) [reply examples/README.md]: Done.

File: examples/research_projects/intel_opts/inference_bf16.py (hunk 2 above)
Comment (hshen14) [reply examples/research_projects/intel_opts/inference_bf16.py:1]: Done. Added some initial ones. Please review. Thanks.

**Comment by hshen14** [comment]
@patrickvonplaten please let me know if you have additional comments.

**Review by anton-l — APPROVED** [review]
Looks nice as a research example! @hshen14 some benchmark results would probably give the examples more visibility, but I'll leave the decision to include them up to you :)

**Comment by hshen14** [comment]
> Looks nice as a research example! @hshen14 some benchmark results would probably give the examples more visibility, but I'll leave the decision to include them up to you :)

Thanks @anton-l. This is on my list to add it later, since BFloat16 support relies on 4th Gen of Intel Xeon Scalable Processor which will be available soon while early preview on AWS.

File: examples/research_projects/intel_opts/README.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+## Diffusers examples with Intel optimizations
```
Comment (patrickvonplaten) [inline examples/research_projects/intel_opts/README.md:1]:
```suggestion
## Diffusers examples with Intel optimizations

**This research project is not actively maintained by the diffusers team. For any questions or comments, please make sure to tag @hshen14 .**
```
```

Comment (hshen14) [reply examples/research_projects/intel_opts/README.md:1]: Done

**Comment by hshen14** [comment]
@patrickvonplaten The notes are updated accordingly per your comments. Please check. Thanks.

**Review by patrickvonplaten — APPROVED** [review]
Thanks a lot!

**Pull request merged**